### PR TITLE
Make some hard-coded JITServer settings configurable

### PR DESCRIPTION
**Pull request by AlexeyKhrabrov**
This PR adds the following `-Xjit:` command line options for JITServer that can be useful for performance tuning:
- `scratchSpaceFactorWhenJITServerWorkload` (default 2) - determines the scratch memory limit for out-of-process compilations at JITServer;
- `reconnectWaitTimeMs` (default 1000 ms) - determines the wait time for the client before attempting to reconnect to the JITServer after a network failure.

The JITServer statistics thread now also uses the value of the existing `timeBetweenPurges` option instead of a hard-coded value.

### CODE REVIEW

**Comment by AlexeyKhrabrov** [comment]
@mpirvu Could you please review this PR? Thanks!

File: runtime/compiler/control/J9Options.hpp [hunk 1]
```diff
@@ -222,8 +222,9 @@ class OMR_EXTENSIBLE Options : public OMR::OptionsConnector
    static int32_t getScratchSpaceFactorWhenJSR292Workload() { return _scratchSpaceFactorWhenJSR292Workload; }
 
 #if defined(J9VM_OPT_JITSERVER)
-   static int32_t getScratchSpaceFactorWhenJITServerWorkload() { return 2; }
-#endif
+   static int32_t _scratchSpaceFactorWhenJITServerWorkload;
+   static int32_t getScratchSpaceFactorWhenJITServerWorkload() { return _scratchSpaceFactorWhenJITServerWorkload; }
```
Comment (mpirvu) [inline runtime/compiler/control/J9Options.hpp:226]: I cannot find where `getScratchSpaceFactorWhenJITServerWorkload` is used.

**Comment by mpirvu** [comment]
@AlexeyKhrabrov just curious, for the new options, have you found in your experiments that other defaults work better?

File: runtime/compiler/control/J9Options.hpp (hunk 1 above)
Comment (AlexeyKhrabrov) [reply runtime/compiler/control/J9Options.hpp:226]: https://github.com/eclipse-openj9/openj9/blob/bdf0b3d46fdf55eb0507a304e85e5f54b8b86628/runtime/compiler/control/CompilationThread.cpp#L8727

**Review by mpirvu — APPROVED** [review]

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit,zlinuxjit jdk17

**Comment by mpirvu** [comment]
jenkins test sanity zlinuxjit jdk17

**Pull request merged**